File: 0001_0100/0073/0073.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    s = f.read().strip()
    d = " abcdefghijklmnopqrstuvwxyz"
    q.write("".join(d[(int(s[i], 27) - (i + 1) % 27 + 27) % 27] for i in range(len(s))))

File: 0001_0100/0073/0073.go
package main

import (
	"fmt"
	"os"
)

func main() {
	f, _ := os.Open("input.txt")
	q, _ := os.Create("output.txt")
	var s string
	fmt.Fscan(f, &s)
	t := []byte(s)
	d := " abcdefghijklmnopqrstuvwxyz"
	for i := range t {
		if t[i] <= '9' {
			t[i] -= '0'
		} else {
			t[i] -= 'A' - 10
		}
		t[i] = d[(int(t[i])-(i+1)%27+27)%27]
	}
	fmt.Fprint(q, string(t))
}
